### ISSUE TITLE
Can't use in rspack when have style block

### ISSUE THREAD

**Issue by Gehbt** [unplugin/unplugin-vue#165]
### Describe the bug

when vue sfc include a style block, plugin will run fail.

![image](https://github.com/user-attachments/assets/9e77b58a-664e-4a6b-8689-d289b5c27b4e)

That will work in webpack.

![image](https://github.com/user-attachments/assets/6aa1a93a-9f5c-4cc3-880b-1ec7d009f947)

### Reproduction

https://github.com/Gehbt/uv-bug-repo

### System Info

```Shell
System:
    OS: Windows 11 10.0.22631
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700H
    Memory: 28.53 GB / 63.67 GB
  Binaries:
    Node: 20.12.2 - ~\AppData\Local\fnm_multishells\16328_1725520498457\node.EXE
    Yarn: 1.22.19 - ~\AppData\Local\fnm_multishells\16328_1725520498457\yarn.CMD
    npm: 10.5.0 - ~\AppData\Local\fnm_multishells\16328_1725520498457\npm.CMD
    pnpm: 9.9.0 - ~\AppData\Local\fnm_multishells\16328_1725520498457\pnpm.CMD
    bun: 1.1.26 - ~\.bun\bin\bun.EXE
  Browsers:
    Edge: Chromium (127.0.2651.74), ChromiumDev (128.0.2677.1)
    Internet Explorer: 11.0.22621.3527
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sxzz** [comment]
Reported it upstream https://github.com/unjs/unplugin/issues/415

**Comment by sxzz** [comment]
Fix & released as `5.1.5`
